### PR TITLE
New lidar object detect optimization

### PR DESCRIPTION
**Pull request by phucla0122**

### CODE REVIEW

**Comment by phucla0122** [comment]
> ### Analysis for commit [f7667d6](https://github.com/CarletonU-2223-SAV/SYSC4907-SAV/commit/f7667d6547a02f7e2e2a71c306ea1d1d49868222) on branch new_lidar_object_detect_optimization
> * Area between target and actual path is 451.87. Target value is 400.00
> * Total objects detected by lidar: 1113
> * Log runtime was 74 seconds. Target value is 60
> 
> Path analysis

Unexpectedly, it decrease

**Comment by phucla0122** [comment]
> ### Analysis for commit [efc6b78](https://github.com/CarletonU-2223-SAV/SYSC4907-SAV/commit/efc6b78dd37778cd9db4aa272bfbf048e54034a0) on branch new_lidar_object_detect_optimization
> * Area between target and actual path is 452.03. Target value is 400.00
> * Total objects detected by lidar: 1581
> * Log runtime was 86 seconds. Target value is 60
> 
> Path analysis

Now it increase??

**Comment by phucla0122** [comment]
> ### Analysis for commit [25f8d62](https://github.com/CarletonU-2223-SAV/SYSC4907-SAV/commit/25f8d6235711a82e8279768ccd6d1aca9b27b5b6) on branch new_lidar_object_detect_optimization
> * Area between target and actual path is 451.39. Target value is 400.00
> * Total objects detected by lidar: 876
> * Log runtime was 69 seconds. Target value is 60
> 
> Path analysis

It decreases

**Review by robellgabriel — COMMENTED** [review]
I noticed that lidar detections have increased throughout the simulation run as the blue lines seem mostly full in color than previous test runs from the old lidar detection. With this said, have you noticed specific improvements with the vehicles performance during the simulation run. From the stats displayed from the integration tests, performance looks almost the same.

File: src/central_control/full_system.launch [hunk 1]
```diff
@@ -15,8 +15,8 @@
 <!-- We're not using this node for now because the sign and car recognition are the same node for now -->
 <!--   <node name="car_detect" pkg="sign_car_recognition" type="car_detect.py" /> -->
   <node name="sign_detect" pkg="sign_car_recognition" type="sign_detect.py" />
-  <node name="object_detect" pkg="lidar_object_recognition" type="lidar_object_detect.py" />
-<!--  <node name="new_object_detect" pkg="lidar_object_recognition_2" type="new_lidar_object_detect.py"/> -->
+  <!--node name="object_detect" pkg="lidar_object_recognition" type="lidar_object_detect.py" /> -->
```
Comment (nicolastuttle) [inline src/central_control/full_system.launch]: If we are scrapping the old object detection, can we just remove this line and rename the new stuff to just `lidar_object_detect` or something? I don't think we should leave files with `new_`

Comment (phucla0122) [reply src/central_control/full_system.launch]: yea, i'll do that. Seems fair

**Comment by phucla0122** [comment]
> I noticed that lidar detections have increased throughout the simulation run as the blue lines seem mostly full in color than previous test runs from the old lidar detection. With this said, have you noticed specific improvements with the vehicles performance during the simulation run. From the stats displayed from the integration tests, performance looks almost the same.

I forgot to say, in some of the run, I forgot to actually run the simulation after changing the metric to produce new NH files. So when I push, it just run the old one, so the stats looks the same in some of the commit

**Comment by robellgabriel** [comment]
> > I noticed that lidar detections have increased throughout the simulation run as the blue lines seem mostly full in color than previous test runs from the old lidar detection. With this said, have you noticed specific improvements with the vehicles performance during the simulation run. From the stats displayed from the integration tests, performance looks almost the same.
> 
> I forgot to say, in some of the run, I forgot to actually run the simulation after changing the metric to produce new NH files. So when I push, it just run the old one, so the stats looks the same in some of the commit

I was talking about the most recent analysis at the time, specifically for commit 0572b54. What's also suprising is that the new lidar detects about half as much compared to the old one, but the blue lines are more full.

**Comment by phucla0122** [comment]
> lking about the most recent analysis at the time, specifically for commit

Like detecting more doesn't means it's better, it just means that the vehicle is more sensitive, which could sometimes be a bad thing since it could cause it to overreact, As min sample increase, the objects that the lidar detect is more concrete so there would be less object being detected, but that also means for each object being detected, it's has a higher guarantee that it's a legit object.

**Comment by nicolastuttle** [comment]
> I was talking about the most recent analysis at the time, specifically for commit [0572b54](https://github.com/CarletonU-2223-SAV/SYSC4907-SAV/commit/0572b54de246d72f44f67e4a95498b3894ae6e2f). What's also suprising is that the new lidar detects about half as much compared to the old one, but the blue lines are more full.

As for lines being more blue with fewer detections, you could have segments where lots is detected, then nothing for a little. This would result in many detections but a relatively white line.

**Review by nicolastuttle — APPROVED** [review]

**Review by robellgabriel — APPROVED** [review]

**Pull request merged**